common/navigator/core/aws.py

```diff
@@ -150,6 +151,67 @@ def move_document(
             new_bucket, os.getenv("AWS_REGION"), new_key or s3_document.key
         )
 
+    def list_files(self, bucket: str, max_keys=1000) -> t.Union[S3Document, bool]:
```

Comment:
@chrisaballard should the return of this method be typed as [typing.Generator](https://docs.python.org/3/library/typing.html#typing.Generator)? I think something like `t.Union[t.Generator[S3Document, None, None], bool]`